client/src/command.ts

```diff
@@ -0,0 +1,147 @@
+};
+
+export const format = (client: LanguageClient) => async (): Promise<void> => {
+  const uri = window.activeTextEditor.document.uri;
+  const version = window.activeTextEditor.document.version;
+  const selections = window.activeTextEditor.selections;
+
+  await client.sendRequest(ExecuteCommandRequest.type, {
+    command: "uroborosql-fmt.executeFormat",
+    arguments: [uri, version, selections],
+  });
+};
+
+export const exportSettings = async (): Promise<void> => {
+  // VSCodeで開いているディレクトリを取得
+  // 開いていない場合はエラーを出して終了
+  const folders = workspace.workspaceFolders;
+  if (folders === undefined) {
+    window.showErrorMessage(
+      "Error: Open the folder before executing this command.",
+    );
+    return;
+  }
+
+  // 設定ファイルのURIを作成
+  const configFile = Uri.joinPath(folders[0].uri, getConfigFileName());
+
+  // VSCode拡張側の設定を取得
+  const vsCodeConfig = workspace.getConfiguration("uroborosql-fmt");
+  const formattingConfig =
+    convertWorkspaceConfigToFormattingConfig(vsCodeConfig);
+
+  // 設定ファイルの設定を取得
+  let existingConfig: OptionsRecord;
+  if (await isFileExists(configFile)) {
+    const file = await workspace.fs.readFile(configFile);
+    existingConfig = JSON.parse(file.toString());
+  } else {
+    // ファイルが存在しなければ設定値なし
+    existingConfig = {};
+  }
+
+  // VSCode 側で明示的に設定したものだけを上書きする
+  const merged = {
+    ...existingConfig,
+    ...objectToSnake(formattingConfig),
+  };
+  const content = JSON.stringify(merged, null, 2);
+
+  const blob: Uint8Array = Buffer.from(content);
+  await workspace.fs.writeFile(configFile, blob);
+};
+
+// uroborosqlfmtrc.json の設定を settings.json に反映
+export const importSettings =
+  (target: ConfigurationTarget) => async (): Promise<void> => {
+    // VSCodeで開いているディレクトリを取得
+    // 開いていない場合はエラーを出して終了
+    const folders = workspace.workspaceFolders;
+    if (folders === undefined) {
+      window.showErrorMessage(
+        "Error: Open the folder before executing this command.",
+      );
+      return;
+    }
+
+    // 設定ファイルのURIを作成
+    const configUri = Uri.joinPath(folders[0].uri, getConfigFileName());
+
+    if (!(await isFileExists(configUri))) {
+      window.showErrorMessage(
+        `Error: Config File is not found: ${configUri.path}`,
+      );
+      return;
+    }
+
+    const blob = await workspace.fs.readFile(configUri);
+    const config: OptionsRecord = JSON.parse(blob.toString());
+
+    // VSCode 拡張の設定を取得
+    const vsCodeConfig: WorkspaceConfiguration =
+      workspace.getConfiguration("uroborosql-fmt");
+
+    // ワークスペース側で設定されている設定項目（そのうち `configurationFilePath` 以外のもの）をすべて null にする
+    const nonNullOptions =
+      convertWorkspaceConfigToFormattingConfig(vsCodeConfig);
+    for (const key of Object.keys(nonNullOptions)) {
+      await vsCodeConfig.update(key, null, target);
+    }
+
+    // 設定ファイルの値で更新する
+    for (const [key, value] of Object.entries(objectToCamel(config))) {
+      await vsCodeConfig.update(key, value, target);
+    }
```

Comment:
現在の書き方だと、update一つ一つをawaitで待機してしまうため、以下のように競合しない更新を並列処理し、それらをまとめてawaitするように処理していただけますか？

```suggestion
    await Promise.all(Object.keys(nonNullOptions).map(key=>vsCodeConfig.update(key, null, target)))

    // 設定ファイルの値で更新する
    await Promise.all(Object.entries(objectToCamel(config)).map(([key, value]) =>vsCodeConfig.update(key, value, target)))
```

Reply:
Promise.all で並列に更新処理をするよう修正しました。

Reply:
変更が確認できません 🤔 commitが漏れていないか確認いただけますか？

Reply:
すみません！リモートに反映するのを忘れていました 🙇